build/common/installer/scripts/fluent-bit-conf-customizer.rb

```diff
@@ -32,6 +32,20 @@ def substituteMultiline(multilineLogging, new_contents)
     return new_contents
 end
 
+def substituteResourceOptimization(resoureceOptimizationEnabled, new_contents)
+  if !resoureceOptimizationEnabled.nil? && resoureceOptimizationEnabled.to_s.downcase == "true"
+    puts "config::Starting to substitute the placeholders in fluent-bit.conf file for resource optimization"
+    if !@os_type.nil? && !@os_type.empty? && @os_type.strip.casecmp("windows") == 0
```

Comment:
i think, we can refactored this logic to function, isWindows and use it since we have multiple instances of the same code in this file.